subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/properties/PropertyValidationAccessTest.groovy

```diff
@@ -83,6 +85,78 @@ class PropertyValidationAccessTest extends Specification {
         ])
     }
 
+    static class TaskWithIterableNested extends DefaultTask {
+        @Nested
+        Iterable<NestedBean> beans
+
+        @Nested
+        List<NestedBean> beanList
+    }
+
+    static class NestedBean {
+        @Input
+        String input
+
+        String nonAnnotated
+    }
+
+    def "analyzes type arguments of Iterables"() {
+        def propertyValidationAccess = new PropertyValidationAccess()
+        def problems = new HashMap<String, Boolean>()
+        when:
+        propertyValidationAccess.collectTaskValidationProblems(TaskWithIterableNested, problems)
+
+        then:
+        problems.keySet() == validationProblems(TaskWithIterableNested, [
+                "property 'beans\$1.nonAnnotated' is not annotated with an input or output annotation",
+                "property 'beanList\$1.nonAnnotated' is not annotated with an input or output annotation"
```

Comment:
This could look like:

```
property 'beans<Iterable>.nonAnnotated' is not annotated with an input or output annotation
property 'beanList<List>.nonAnnotated' is not annotated with an input or output annotation
```

...or something.